xc/xenops_server_xen.ml

```diff
@@ -251,17 +251,7 @@ let device_kind_of_backend_keys backend_keys =
   with Not_found -> Device_common.Vbd !Xenopsd.default_vbd_backend_kind
 
 let params_of_backend backend =
-  let blockdevs, files, xendisks, nbds =
-    List.fold_left
-      (fun (blockdevs, files, xendisks, nbds) ->
-         function
-         | BlockDevice b -> (b::blockdevs, files, xendisks, nbds)
-         | File f -> (blockdevs, f::files, xendisks, nbds)
-         | XenDisk x -> (blockdevs, files, x::xendisks, nbds)
-         | Nbd nbd -> (blockdevs, files, xendisks, nbd::nbds))
-      ([], [], [], [])
-      backend.implementations
-  in
+  let xendisks, blockdevs, files, nbds = Storage_interface.implementations_of_backend backend in
```

Comment:
I was worried that the order is easy to get wrong here, but the types of each of these are different, so should be fine.

Reply:
Yep, I've also double-checked the types by hand